### ISSUE TITLE
`get_image_details': undefined method `name' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by isamuelson** [Azure/vagrant-azure#122]
.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/run_instance.rb:153:in `get_image_details': undefined method`name' for nil:NilClass (NoMethodError)
    from /Users/ido/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/run_instance.rb:63:in `block in call'
    from /Users/ido/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/util/timer.rb:7:in`time'
    from /Users/ido/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/run_instance.rb:62:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/synced_folders.rb:87:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/provision.rb:80:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in`busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/call.rb:53:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /Users/ido/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/connect_azure.rb:23:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/box_check_outdated.rb:36:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/config_validate.rb:25:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builtin/handle_box.rb:56:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/builder.rb:116:in`call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/util/busy.rb:19:in`busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:224:in`action_raw'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:199:in `block in action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/environment.rb:561:in`lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/machine.rb:185:in`action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.1/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'

**Comment by devigned** [comment]
Looks like we couldn't find the latest version given the location, publisher, offer and sku. See the code below.

``` ruby
latest = azure.compute.virtual_machine_images.list(location, publisher, offer, sku).value!.body.last
azure.compute.virtual_machine_images.get(location, publisher, offer, sku, latest.name).value!.body
```
via: https://github.com/Azure/vagrant-azure/blob/v2.0/lib/vagrant-azure/action/run_instance.rb#L152-L153

What are the values for these parameters?

We should not fail so badly here and should provide an error message to someone that we couldn't find the image for the given parameters.

**Comment by gbhatnag** [comment]
Having this same issue with Vagrant + Azure set up on Mac OS X.  I've tried to use specific URN strings based on the list generated by `azure vm image list` including `b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_4-LTS-amd64-server-20160516-en-us-30GB` as well as the shorter `Ubuntu-14_04_4-LTS-amd64-server-20160516-en-us-30GB`.  The same `<provider>:<offer>:<version>` structure is not available with the latest version of azure CLI.  Below is environment info + the stack trace.

We need to set up VMs in the new Azure Portal using Vagrant from remote machines.  Been working on this for a while and would appreciate help at your earliest convenience.

**Environment**
Mac OS X 10.11
Vagrant 1.8.4
vagrant-azure 2.0.0.pre1
azure CLI 0.10.1

**Stack Trace**

```
$ vagrant up --provider=azure
Bringing machine 'default' up with 'azure' provider...
==> default: Launching an instance with the following settings...
==> default:  -- Management Endpoint: https://portal.azure.com
==> default:  -- Subscription Id: [snipped]
==> default:  -- Resource Group Name: [snipped]
==> default:  -- Location: westus
==> default:  -- VM Name: [snipped]
==> default:  -- VM Size: Standard_DS11_v2
==> default:  -- Image URN: canonical:ubuntuserver:16.04.0-DAILY-LTS:latest
/Users/gauravb/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/run_instance.rb:153:in `get_image_details': undefined method `name' for nil:NilClass (NoMethodError)
    from /Users/gauravb/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/run_instance.rb:63:in `block in call'
    from /Users/gauravb/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/util/timer.rb:7:in `time'
    from /Users/gauravb/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/run_instance.rb:62:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/synced_folders.rb:87:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/provision.rb:80:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /Users/gauravb/.vagrant.d/gems/gems/vagrant-azure-2.0.0.pre1/lib/vagrant-azure/action/connect_azure.rb:23:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/box_check_outdated.rb:36:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:225:in `action_raw'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:200:in `block in action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/environment.rb:561:in `lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:186:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:186:in `action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

**Comment by gbhatnag** [comment]
Additionally, `azure vm image list-publishers` and `azure vm image list-offers` does not work with the latest `azure 0.10.1`, so I cannot know for sure how to structure the URN and, if I leave it blank, I get the above error.

Here's what the output of `azure vm image list` looks like (it's just a few rows out of many):

```
$ azure vm image list
info:    Executing command vm image list
+ Fetching VM images                                                           
data:    Name                                                                                                                  Category  OS       Publisher                                          
data:    --------------------------------------------------------------------------------------------------------------------  --------  -------  ---------------------------------------------------
data:    03f55de797f546a1b29d1b8d66be687a__CoreCLR-x64-Beta5-Linux-PartsUnlimited-Demo-App-201504.29                           Public    Linux    Microsoft Visual Studio Group                      
data:    03f55de797f546a1b29d1b8d66be687a__Team-Foundation-Server-2013-Update4-WS2012R2                                        Public    Windows  Microsoft Visual Studio Group                      
data:    03f55de797f546a1b29d1b8d66be687a__Team-Foundation-Server-2013-Update4-WS2012R2-201503.20                              Public    Windows  Microsoft Visual Studio Group                      
data:    03f55de797f546a1b29d1b8d66be687a__Team-Foundation-Server-2013-Update4-WS2012R2-201503.27                              Public    Windows  Microsoft Visual Studio Group                      
...
data:    b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_4-LTS-amd64-server-20160314-en-us-30GB                                 Public    Linux    Canonical                                          
data:    b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_4-LTS-amd64-server-20160406-en-us-30GB                                 Public    Linux    Canonical                                          
data:    b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_4-LTS-amd64-server-20160509.1-en-us-30GB                               Public    Linux    Canonical                                          
data:    b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_4-LTS-amd64-server-20160516-en-us-30GB                                 Public    Linux    Canonical                                          
...
```

**Comment by gbhatnag** [comment]
I've limited our `Vagrantfile` to only the mandatory config fields listed in your docs at https://github.com/Azure/vagrant-azure#configuration

The current version of the plugin is unusable for all users.  Can you provide docs, registry or a list of URNs that we can use to avoid this error?

**Comment by devigned** [comment]
This should be fixed in the 2.0.0.pre2 release. Please reopen if it's still an issue.
